### ISSUE TITLE
TypeScript compilation errors when using "noImplicitAny"

### ISSUE THREAD

**Issue by ulrichb** [nestjs/nest#1354]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

TSC emits the following errors when having --noImplicitAny enabled (or --strict) and having *no* --skipLibCheck switch.

```plain
node_modules/@nestjs/common/interfaces/external/serve-static-options.interface.d.ts(59,19): error TS7006: Parameter 'res' implicitly has an 'any' type.
node_modules/@nestjs/common/interfaces/features/pipe-transform.interface.d.ts(5,30): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/common/interfaces/websockets/web-socket-adapter.interface.d.ts(4,47): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/common/interfaces/websockets/web-socket-adapter.interface.d.ts(5,51): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/guards/guards-consumer.d.ts(6,86): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/guards/guards-consumer.d.ts(7,65): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/guards/guards-context-creator.d.ts(11,45): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/helpers/context-creator.d.ts(5,86): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/helpers/context-creator.d.ts(7,41): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/helpers/external-context-creator.d.ts(33,88): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/helpers/external-context-creator.d.ts(37,32): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/helpers/external-context-creator.d.ts(37,63): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/injector/module.d.ts(17,18): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/interceptors/interceptors-consumer.d.ts(6,94): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/interceptors/interceptors-consumer.d.ts(7,65): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/interceptors/interceptors-context-creator.d.ts(10,45): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
node_modules/@nestjs/core/pipes/pipes-consumer.d.ts(10,9): error TS7008: Member 'metatype' implicitly has an 'any' type.
node_modules/@nestjs/core/pipes/pipes-consumer.d.ts(11,9): error TS7008: Member 'type' implicitly has an 'any' type.
node_modules/@nestjs/core/pipes/pipes-consumer.d.ts(12,9): error TS7008: Member 'data' implicitly has an 'any' type.
node_modules/@nestjs/core/pipes/pipes-context-creator.d.ts(10,45): error TS7019: Rest parameter 'args' implicitly has an 'any[]' type.
```

## Expected behavior
No tsc errors.

## Minimal reproduction of the problem with instructions
See above

## What is the motivation / use case for changing the behavior?
Enabling --noImplicitAny in consuming applications.

**Comment by marcus-sa** [comment]
You should exclude node_modules

**Comment by ulrichb** [comment]
@marcus-sa: I don't think that this works. TS must parse all declarations (even the internal lib.*.d.ts declarations). And --skipLibCheck just drops the type checking errors.

IMO this should be solved by adding `any` to the concerned types (like [here](https://github.com/nestjs/nest/blob/c6ec8878e0864469fa5e83d97d3ba07df397b09e/packages/common/interfaces/external/serve-static-options.interface.ts#L68)).

I could also propose a PR.

**Comment by brianmcd** [comment]
As a workaround, you can set `skipLibCheck: true` in your `tsconfig.json`.

**Comment by kamilmysliwiec** [comment]
It's already fixed in 6.0.0 branch and will be published together with nearest major release

**Comment by ulrichb** [comment]
@kamilmysliwiec Nice. Thanks!
